cmd/credentialUtil.go

```diff
@@ -113,10 +109,17 @@ func getBlobCredentialType(ctx context.Context, blobResourceURL string, canBePub
 			})
 
 		isContainer := copyHandlerUtil{}.urlIsContainerOrVirtualDirectory(resourceURL)
-		isPublicResource := false
+		isPublicResource = false
 
 		if isContainer {
 			containerURL := azblob.NewContainerURL(*resourceURL, p)
+			bURLparts := azblob.NewBlobURLParts(*resourceURL)
+
+			if bURLparts.ContainerName == "" || strings.Contains(bURLparts.ContainerName, "*") {
+				// Service level searches can't possibly be public.
+				return false
+			}
+
 			if _, err := containerURL.GetProperties(ctx, azblob.LeaseAccessConditions{}); err == nil {
```

Comment:
>containerURL.GetProperties [](start = 16, length = 26)

the blobs may be public but the container may not be. perhaps test with such a scenario to confirm behavior

Reply:
That scenario already is tested. You can't list blobs if a container isn't public, thus it would have to be a direct link to a blob, which we'd attempt to get properties to. I'll test this regardless though.